packages/eslint-plugin/src/rules/no-var-requires.ts

```diff
@@ -20,14 +24,33 @@ export default util.createRule<Options, MessageIds>({
     messages: {
       noVarReqs: 'Require statement not part of import statement.',
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowPackageDotJson: {
+            type: 'boolean',
+          },
+        },
+      },
+    ],
   },
-  defaultOptions: [],
+  defaultOptions: [{ allowPackageDotJson: false }],
   create(context) {
+    const allowPackageDotJson = context.options[0]?.allowPackageDotJson;
     return {
       'CallExpression, OptionalCallExpression'(
         node: TSESTree.CallExpression | TSESTree.OptionalCallExpression,
       ): void {
+        if (
+          allowPackageDotJson &&
+          node.arguments[0]?.type === AST_NODE_TYPES.Literal &&
+          typeof node.arguments[0]?.value === 'string' &&
+          /package.json$/.test(node.arguments[0].value)
```

Comment:
use `endsWith` instead of a regex.
Also prefix with a `/` so that it doesn't false positive on things like `./mypackage.json`.

```suggestion
          node.arguments[0].value.endsWith('/package.json'),
```